File: mypay.sources/mypay4-fe/projects/mypay4-fe-operatore/src/app/model/giornale.ts
/*
 *     MyPay - Payment portal of Regione Veneto.
 *     Copyright (C) 2022  Regione Veneto
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU Affero General Public License as
 *     published by the Free Software Foundation, either version 3 of the
 *     License, or (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU Affero General Public License for more details.
 *
 *     You should have received a copy of the GNU Affero General Public License
 *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
import { DateTime } from 'luxon';
import { MapperDef, MapperType } from 'projects/mypay4-fe-common/src/public-api';

export class Giornale {
  public static readonly MAPPER_S2C_DEF = [
    new MapperDef(MapperType.DateTime,'dataOraEvento','local-date-time'),
  ];

  mygovGiornaleId: number;
  dataOraEvento: DateTime;
  idDominio: string;
  iuv: string;
  evento: string;
  idPsp: string;

  tipo: string;
  sottotipo: string;
  categoria: string;
  esito: string;
  contestoPagamento: string;
  tipoVersamento: string;
  componente: string;
  idFruitore: string;
  idErogatore: string;
  idStazione: string;
  canalePagamento: string;
  parametriInterfaccia: string;

  details: object[];
}

File: mypay.sources/mypay4-be/src/main/java/it/regioneveneto/mygov/payment/mypay4/model/common/Giornale.java
/**
 *     MyPay - Payment portal of Regione Veneto.
 *     Copyright (C) 2022  Regione Veneto
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU Affero General Public License as
 *     published by the Free Software Foundation, either version 3 of the
 *     License, or (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU Affero General Public License for more details.
 *
 *     You should have received a copy of the GNU Affero General Public License
 *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
package it.regioneveneto.mygov.payment.mypay4.model.common;

import it.regioneveneto.mygov.payment.mypay4.dto.BaseTo;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.util.Date;

@Data
@NoArgsConstructor
@SuperBuilder(toBuilder=true)
public abstract class Giornale extends BaseTo {

  private Long mygovGiornaleId;
  private int version;
  private Date dataOraEvento;
  private String identificativoDominio;
  private String identificativoUnivocoVersamento;
  private String codiceContestoPagamento;
  private String identificativoPrestatoreServiziPagamento;
  private String tipoVersamento;
  private String componente;
  private String categoriaEvento;
  private String tipoEvento;
  private String sottoTipoEvento;
  private String identificativoFruitore;
  private String identificativoErogatore;
  private String identificativoStazioneIntermediarioPa;
  private String canalePagamento;
  private String parametriSpecificiInterfaccia;
  private String esito;
}
